src/lib.rs

```diff
@@ -201,21 +201,19 @@ apps/ @octocat
 
         let co = parse(example).unwrap();
         assert_eq!(co.rules.len(), 7);
+        
+        dbg!(co.rules.iter().map(|rule| &rule.pattern).enumerate().collect::<Vec<_>>());
 
-        macro_rules! test_rule {
-            ($index:expr, $pattern:expr, $owners:expr, $test:expr) => {
-                let pattern = $pattern;
-                let owners = $owners;
-                let rule = &co.rules[$index];
-                assert_eq!(rule.pattern, pattern);
-                assert_eq!(rule.owners, owners);
-
-                let m = co.matches($test);
-                assert_eq!(m, Some(rule));
-            };
-        }
+        let test_rule = |index: usize, pattern: &str, owners: Vec<Owner>, test: &str| {
```

Comment:
Why do you prefer it this way? We will not get proper line numbers on assertion errors in this case.

Reply:
I precisely did this in order to get better backtraces than with the macro :D (also I had issues with type errors during refactoring and the compiler messages were rather annoying because of the macro)